database/src/main/postgres/runs/V1.8.3__get_partitioning_checkpoints.sql

```diff
@@ -28,53 +28,69 @@ CREATE OR REPLACE FUNCTION runs.get_partitioning_checkpoints(
+-- Function: runs.get_partitioning_checkpoints(4)
+--      Retrieves checkpoints (measures and their measurement details) related to a
 --      given partitioning (and checkpoint name, if specified).
 --
 -- Parameters:
 --      i_partitioning          - partitioning of requested checkpoints
---      i_limit                 - (optional) maximum number of checkpoint's measurements to return
---                                if 0 specified, all data will be returned, i.e. no limit will be applied
+--      i_limit                 - (optional) maximum number of checkpoints to return
+--                                i_limit relates to amount of checkpoints returned, not the amount of rows returned
+--                                as usually there is more than one row per checkpoint
+--      i_offset                - (optional) offset of the first checkpoint to return
+--      i_checkpoint_name       - (optional) name of the checkpoint
+
+-- Note: i_limit and i_offset are used for pagination purposes;
+--       checkpoints are ordered by process_start_time in descending order
+--       and then by id_checkpoint in ascending order
 --
 -- Returns:
---      i_checkpoint_name       - (optional) if specified, returns data related to particular checkpoint's name
 --      status                  - Status code
 --      status_text             - Status message
 --      id_checkpoint           - ID of the checkpoint
 --      checkpoint_name         - Name of the checkpoint
 --      author                  - Author of the checkpoint
---      measuredByAtumAgent     - Flag indicating whether the checkpoint was measured by ATUM agent
+--      measured_by_atum_agent  - Flag indicating whether the checkpoint was measured by ATUM agent
 --      measure_name            - Name of the measure
 --      measure_columns         - Columns of the measure
 --      measurement_value       - Value of the measurement
 --      checkpoint_start_time   - Time of the checkpoint
 --      checkpoint_end_time     - End time of the checkpoint computation
+--      has_more                - Flag indicating whether there are more checkpoints available
 --
 -- Status codes:
 --      11 - OK
 --      41 - Partitioning not found
+--      42 - No checkpoint data found
 --
 -------------------------------------------------------------------------------
-DECLARE
-    _fk_partitioning                    BIGINT;
+$$
 BEGIN
-    _fk_partitioning = runs._get_id_partitioning(i_partitioning);
-
-    IF _fk_partitioning IS NULL THEN
+    PERFORM 1 FROM runs.partitionings WHERE id_partitioning = i_partitioning_id;
+    IF NOT FOUND THEN
         status := 41;
         status_text := 'Partitioning not found';
         RETURN NEXT;
         RETURN;
     END IF;
 
     RETURN QUERY
+        WITH limited_checkpoints AS (
+            SELECT C.id_checkpoint,
+                   C.process_start_time,
+                   ROW_NUMBER() OVER (ORDER BY C.process_start_time DESC, C.id_checkpoint) AS rn
```

Comment:
This is a really cool trick. 😎 
But I think it would be more efficient to have a pre-query
```sql
IF i_limit IS NOT NULL THEN
    SELECT count(1) > i_limit
    FROM runs.checkpoints C
    WHERE C.fk_partitioning = i_partitioning_id
        AND (i_checkpoint_name IS NULL OR C.checkpoint_name = i_checkpoint_name)
    ORDER BY C.process_start_time DESC, C.id_checkpoint
    LIMIT i_limit + 1 OFFSET i_offset
    INTO _has_more;
ELSE
    _has_more := false;
END IF;
```
And then use  the `_has_more` in the query. The query will be actually cached by the PG engine.